speechbrain/utils/parameter_transfer.py

```diff
@@ -285,13 +334,18 @@ def load_collected(self):
             if not self.is_loadable(name):
                 continue
             filename = name + PARAMFILE_EXT
-            paramfiles[name] = self.collect_in / filename
 
             if name in self.is_local:
                 logger.debug(
-                    f"Redirecting (loading from local path): {paramfiles[name]} -> {self.paths[name]}"
+                    f"Redirecting (loading from local path): {name} -> {self.paths[name]}"
                 )
                 paramfiles[name] = self.paths[name]
+            elif self.collect_in is not None:
+                paramfiles[name] = self.collect_in / filename
+            else:
+                raise ValueError(
+                    f'Pretrainer could not locate file `{name}` for loading! (hint: `Pretrainer.collect_in` no longer defaults to "model_checkpoints", is your code expecting that file to be there?)'
+                )
```

Comment:
Am I right in understanding that this could only be reached if `collect_in` is not called first? And since `model_checkpoints` is the default for `collects_in` I guess this would happen only if the code had been run once with `collects_in` and then changed to not call this any more. Makes sense that there's a message here, and this could be a helpful message, but just wanted to clarify the scenario we were trying to catch here.

Reply:
That would be correct, actually. I wrote this warning with incorrect assumptions. Will double check the code paths to understand what can exactly trigger this.

Reply:
Well, more specifically, `collect_in` now defaults to `None` with this PR, but `collect_files` should presumably still be called by the user either way?